File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_py/code_output/PersonRequest.py
import re


class PersonRequest:
    def __init__(self, name: str, sex: str, phone_number: str):
        self._name = self._validate_name(name)
        self._sex = self._validate_sex(sex)
        self._phone_number = self._validate_phone_number(phone_number)

    @staticmethod
    def _validate_name(name):
        if name is None or name == "" or len(name) > 33:
            return None
        return name

    @staticmethod
    def _validate_sex(sex):
        if sex is None or sex not in ("Man", "Woman", "UGM"):
            return None
        return sex

    @staticmethod
    def _validate_phone_number(phone_number):
        if phone_number is None or phone_number == "" or len(phone_number) != 11 or not re.fullmatch(r"\d{11}", phone_number):
            return None
        return phone_number

    def get_name(self):
        return self._name

    def get_sex(self):
        return self._sex

    def get_phone_number(self):
        return self._phone_number

File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_cpp/code_output/PersonRequest.cpp
#include <optional>
#include <regex>
#include <string>

class PersonRequest {
private:
    std::optional<std::string> name_;
    std::optional<std::string> sex_;
    std::optional<std::string> phoneNumber_;

    static std::optional<std::string> validateName(const std::optional<std::string>& name) {
        if (!name.has_value() || name->empty() || name->length() > 33) {
            return std::nullopt;
        }
        return *name;
    }

    static std::optional<std::string> validateSex(const std::optional<std::string>& sex) {
        if (!sex.has_value() ||
            (*sex != "Man" && *sex != "Woman" && *sex != "UGM")) {
            return std::nullopt;
        }
        return *sex;
    }

    static std::optional<std::string> validatePhoneNumber(const std::optional<std::string>& phoneNumber) {
        if (!phoneNumber.has_value() || phoneNumber->empty() ||
            phoneNumber->length() != 11 ||
            !std::regex_match(*phoneNumber, std::regex("\\d{11}"))) {
            return std::nullopt;
        }
        return *phoneNumber;
    }

public:
    PersonRequest(const std::optional<std::string>& name,
                  const std::optional<std::string>& sex,
                  const std::optional<std::string>& phoneNumber)
        : name_(validateName(name)),
          sex_(validateSex(sex)),
          phoneNumber_(validatePhoneNumber(phoneNumber)) {}

    const std::optional<std::string>& getName() const {
        return name_;
    }

    const std::optional<std::string>& getSex() const {
        return sex_;
    }

    const std::optional<std::string>& getPhoneNumber() const {
        return phoneNumber_;
    }
};